### ISSUE TITLE
OS CHECKLIST

### ISSUE THREAD

**Issue by Giuliog123** [Giuliog123/OS-CheckList#1]
Check Non functional requirements 
appendix : Technological Gap for elderly users  
Use my mybib for refenrences 
WCAG for keyboard 

Task 1 Checklist
Design Documentation 
Proposal
**- Business context**
In the business context, you need to describe the business e.g. Our client, xxxxxxxxxxxx (client company name) is a well stablished business that is aiming to xxxxxxxxxxxx (company objectives). The client is looking or a digital solutions to help xxxxxxxxx, xxxxxxx, xxxxxx (what does it really want to change with the new digital solution).

**Stakeholders** 
search most common types of  stakeholders and paste time and date of the search into a research log. 
**Constraints** 
search most common types of  constraints and paste time and date of the search into a research log.  Examples are high cost, maintenance and to set it up.
**- empathy map**
An empathy map is a 2x4 grid that contains speaks, thinks, does and feels. This will aim to capsulise and analyse user preferences and requirements to adjust the digital solution into the best quality possible.

In speaks we are aiming to collect direct information from the customer such as I need more inform from the local community or what upcoming events are in my local area ?. In thinks, we are to understand user thinking such as I wish there was an interactive website where I could see all upcoming events in the local area and it would be so much easier if there was a posting blog. 

In does, we aim to collect information real actions on location and contact info from the business such as I came to visit but the centre was fully booked and I came to visit but I found out that it was actually closed. And in feels we are to understand how the user feels like happy, confused, stressed, annoyed and overwhelmed.

- user stories
In user stories we aim to develop a 4x6 grid which involves As a [user], I want..., So that.... and Acceptance criteria in the top 4 headers of the grid. On the left headers of the grid, we aim to implement first time customers, vulnerable groups, loyal customers, and 2 more depending on the task. 

**- Potential Users:**

I need to identify additional potential users and unique needs to ensure that the client has the boarder audience as possible. 
Such as elderly users ( difficulty with technology so they need simple interface and layout), investors (insights into market trends and investment opportunities), Technicians ( Scheduling and management tool to allow service providers to efficiently handle installtions and maintenance requests) Government ( enable them to access reports ) community leaders and non profit organisations ( enable them for planning and executing large scale sustainability projects). 

**Personas** 
I need a one paragraph description of the personas such as these personas will give insight into users behaviours, motivations and challenges. 
A persona example
Person description such as profession role and what type of customer is it such as eco conscious homeowner.
A bio descriping the person such as passioante about going to the zoo and xxxxxx. 
About personality, introvert vs extrovert, analytical vs creative, busy vs time rich, messy vs organised, passive vs active. 
About interests (sustainable living) , influences (friendds and family and industry news), goals( reduce household energy costs), needs and expectations ( access to clear information on green energy products), motivations (Wants to contribute to a more sustaiblable world for their children) and pain points(high upfront costs to install) . 

If i have time i can add another persona about a small size business affected by this. 

**- functional requirements**
In functional requirements, start with Homepage and describe it. 
User management & Example given by the exam such as Consultation and Installation scheduling
Products Information & Accessibility Features
Data analytics and reporting
Secure payment and subscription service
Collaboration tools & Integration with external systems.

**- non functional requirements**

Performance & Scalability
Security & Data Protection
Usability & Accessibility
Reliability & Availability 
Integration Capabilities 
Support 

**- decomposition**
I would like to implement a decomposition diagram as it helps to by breaking down complex system into manageable small sections. There are several benefits such as Clairrty in system architecture, improved development workflow and enhanced user experience design. Use figma to create the decomposition diagram.  

**Key performance Indicator (KPI) and User Acceptance Criteria** 

Key performance Indicator (KPI) and User Acceptance Criteria 
Customer Satisfaction and Support 
Net Promoter Score (NPS) - Give short description
Customer Support Response Time  - Give short description
User Feedback - Give short description

Accessibility and Inclusivity 
Usage of accessibility features - Give short description
feedback from accessibility users - Give short description

System performance and Reliability
Platform Uptime Percentage - Give short description
Page load speed - Give short description
Bug resolution time - Give short description

User acceptance criteria 
User registration and login 
Accessibility Features 
2 user acceptance criteria from the exam such as Consultation and Installation booking 
Carbon footprint and energy tracking
Performance and Security 
Customer support and feedback 

**- description**
I need to implement a description of the proposed solution such as describing different emerging trends that enhances its platform and provide solutions for users such as:
Ai powered optimisation
IoT integration
Augemented Reality (AR) 
Incentives 
Enhanced accessibility features such as ai driven voice assitance 

Now talk about how this will be compatible with the exam requirements like how will iot integration be compatible with devices for green energy. 
1) IoT integration compability with external devices 
2) Cloud based data (users can connect to their account by any device) storage and security 
3)Third party software.

Now talk about key features such as accessibiltiy, sustainability considerations and inclusion and usability

- justification
How the recommended solution meets the needs of the client
how potential risks will be mitigated
regulatory guidelines and legal requirements will be addressed

- Appendix 1
Virtual reality 
Ai chatbots
Payment 
Bloackchain for secure and transparent transactions for emerging technologies

- Appendix 2
Business Context
 Client Brief Review
Interested in digitalisation and why
Client problems
What would they want in the solution
"I am proposing to develop... "
 Stakeholders
Understand who we are working with and effecting
Tailored and specialised to an audience
Think:
Media
Staff
Clients
Community
 Constraints and Scope
Whys listing constraints and scopes are important
Scope Creep
Budget, Time, Technology, Compatibility, Laws
Features and Functionality
 Empathy Mapping
Think about:
What customers care about
Challenges they face
What motivates them
How the platform helps
Says, Thinks, Does, Feels
Without the solution
"I need information"
"Could I..." "So that..."
"I wish..."
"It would be easier if they had {feature}"
Personalised Experience
 SWOT Analysis
Strength
Weakness
Opportunities
Threats
 User Stories & Acceptance
As a {user}..., I want..., So that...
Acceptance Criteria (features that we will include in the solution)
Find information
Compare with competitors
Easier management
 Functional & Non Functional Requirements

What the system must do, performance, security, usability, maintainability
Can sort by page or list them
Bycrypt
Secure Cookies
Validation
User Inputs requirement
 Decomposition of Requirements

Define Decomposition and why you are doint it
Claity In System Architecture - how components interact
Improved Dev workflow
User Experience - ensure all is logical sturcture
Decompose each page or a requirements
Could do like a tree thing with all the pages and what they would contain
 KPI's & User Acceptance

Reviews
Customer satisfaction rate (surveys)
Number of ...
Repeat visitors
Revenue growth
Page load time
Bounce rate (how many users interact with the site)
Proposed Solution
 Description of Proposed Solution
What the website will include
Features
Experience
Justification
 Meet the Client and User needs

Mention what the client will see and do
How it helps them and us
What problems it solves
 Risks, Mitigation & Wider Issues

Information inaccuracy
Miscommunication
Data loss
Data breaches
 Regulatory Guidelines and Legal Requirements

GDPR - If website is offering goods or services to those located in the European Union, Iceland, Norway, Lichtenstien, Switzerland or UK you must comply with sweeping privacy laws generated by the GDPR. Complying starts with privacy policy that details what, how, when, and where data is collected while also allowing users to access, rectify, and delete their data.
COPPA - Protects children's privacy and keep them safe online. COPPA requires that websites obtain consent from parents before collecting personal information from kids under the age of 13
EU Cookie Law - Requires websites to have a dedicated cookie policy and to get consent from users before they can store or retrieve personal information on a compute, smartphone, or tablet. Allows for an informed choice regarding whether or not they should continue providing the information.
Accessibility - Compliance with WCAG to ensure usability for individuals with disabilities. Screen reader compatibility, text to speech options, high contrast colours, adjustable font.
Intellectual Property - Compliance with Copyright, Designs and Patents Act 1988. Ensure all content used is owned by the client or properly licensed.
Online Saftey - Online saftey features in line with the Online Saftey Bill to protect users from harmful content. Moderation, reporting mechanisms, clear guidlines on acceptable use.
NCSC guidlines (Cyber Security) - Encryption, secure authentication, regular security audits, incident response plans.
Task 1B
**- Visual designs** 
- Logo
EST.   1990 
https://coolors.co/

show where you created it from, the inspiration behind the logo, describe how you feel the audience will feel like childish or too luxurious. 

**- Colour Scheme**
Demonstrate how there was a research in the industry and talk about all the competitors logo types, talk about the colours being prominent in the industry and talk about the colour palate you came up with using collors.co. The following will dependent on the colour pallet numbers. 
With #
Array
Object 
talk about how you were constantly checking the contrasts and colours for compliance with WCAG regulations using WebAim website. Talk about the initial ideas and Different ideas bellow. 

**- Navigation Bar** 
Attach and explain the designs made using figma.
**- Home Page** 
Paste a screenshot of the home page design 
**- About Page** 
Paste a screenshot of the about page design 
- **Educational Visits Page**
Paste a screenshot of the Educational Visits page design 
**- Booking System**
Paste a screenshot of the booking system page design 
**- Login/Register**
Paste a screenshot of the login/register page design 
- Full Solution
Paste a screenshot of the full solution page design 
- Data Requirements 
- ERD + Data Flow 
- In ERD, the structure is many to many and one to one. For example ' cashier is many to many with items list and category is many to many with items list. 
- ERD 
- Data Dictionary 
- data disctionary needs a column, type, null, key and description. 
- Algorithm 
- Wireframes and the flow 
- Booking system Flowchart, view - full flowchart 
- Booking system flowchart, view closer on main logic
- Test Strategy
This should have Date of test, component to be tested, Type of test to be carried out, Prerequisities and dependencies. 
Functional Testing 
Registration form - functional testing 
Login form- functional testing 
purchase form - - functional testing 
password recovery - functional testing 

Integration Testing
Payment Gateway Integration
IoT Device Integration
Calendar Service Integration

User Experience Testing 
Usabilility Testing 
Accessibility Testing 

Performance Testing 
Page laod Time 
Response Time 
Scalability
